packages/back-end/src/controllers/stripe.ts

```diff
@@ -80,7 +82,28 @@ export async function postNewSubscription(
     ],
     success_url: `${APP_ORIGIN}/settings/team?org=${org.id}&subscription-success-session={CHECKOUT_SESSION_ID}`,
     cancel_url: `${APP_ORIGIN}${returnUrl}?org=${org.id}`,
-  });
+  };
+
+  if (startFreeTrial) {
+    payload.subscription_data = {
+      trial_period_days: 14,
+      trial_settings: {
+        end_behavior: {
+          missing_payment_method: "cancel",
+        },
+      },
+    };
+    payload.payment_method_collection = "if_required";
+  }
+
+  const session = await stripe.checkout.sessions.create(payload);
+
+  if (startFreeTrial) {
+    await updateOrganization(org.id, {
+      freeTrialDate: new Date(),
+    });
```

Comment:
If they click through to Stripe, but don't actually start the trial, will this stop them from starting a trial in the future?

Reply:
good call, moving the update to the success endpoint